### ISSUE TITLE
Ambiguous Peptides b/c of different mod names for the same mod

### ISSUE THREAD

**Issue by blfrey** [smith-chem-wisc/MetaMorpheus#819]
Rob and I discovered this issue.
Here is an example of MM calling a peptide ambiguous, but it is really the same exact peptide:
AGVMGPP[Mod:Oxidation on P]GSR or AGVMGPP[UniProt:4-hydroxyproline]GSR
Oxidation of proline is the same as 4-hydroxyproline.
I wonder if this same issue occurs with other PTMs (e.g. acetyl searched as a variable mod, but also appearing in the UniProt xml in places).
Rob and I thought one way to handle this might be to make a "synonyms list" somewhere (mzlib?). Then you could check that list to see if the peptide was really ambiguous.  Also, there could be a hierarchy in the list that tells which one to list (e.g. we'd want the UniProt one to be listed rather than the variable oxidation of P).

**Comment by stefanks** [comment]
Yes, this is an issue with phosphorylations as well. The problem is exacerbated here, because they are not strictly "synonyms":  phosphorylations in MetaMorpheus have the additional information about neutral losses encoded, while UniProt mods do not. So PEPS[MetaMorpheus phospho]PEP might have a higher score than PEPS[UniProt phospho]PEP

**Comment by zrolfs** [comment]
How does this even happen? Why would GPTMD find a novel mod if it was already annotated in the database? Shouldn't the annotated mod outscore the unmodified with a notch?

**Comment by stefanks** [comment]
phospho in uniprot ptmlist does not include information on possible neutral losses

**Comment by zrolfs** [comment]
But oxidation isn't considered as possible neutral loss, and neither is the acetyl I ran into.

**Comment by stefanks** [comment]
Sometimes the annotated mod does not outscore the unmodified with a notch.. even though the modification is really there

**Comment by trishorts** [comment]
Two points to add. 
1.) if there are no fragments that can distinguish between a previously annotated mod and a gptmod, then we should just go with the annotated mod.
2.) fragments that can distinguish include: a.) fragment with annotated mod; b.) fragment from unmodified where there should have been an annotated mod; c.) fragment with mod at position made possible by gptmd.

**Comment by rmillikin** [comment]
I've added a change to GPTMD in #1637 that prevents GPTMD modifications from being annotated in the database if there is already a UniProt modification with the same mass (within 0.005 Da) on the same residue. This should prevent UniProt-MM mod ambiguity in the vast majority of cases. 

However, we may want to revisit this in the case of glycosylations because many of these mods are isobaric. Meaning that in many cases we'd want to annotate a glyco even if there is already a glyco with the same mass on that residue. This is probably somewhat not problematic currently because UniProt doesn't have a lot of annotated glycosylations, but eventually we'll probably want to address it.
